src/deepsparse/pipeline.py

```diff
@@ -730,6 +805,34 @@ def route_input_to_bucket(
         pass
 
 
+def _initialize_executor_and_workers(
+    batch_size: Optional[int],
+    workers_or_executor: Optional[Union[int, ThreadPoolExecutor]],
+) -> Tuple[Optional[ThreadPoolExecutor], int]:
```

Comment:
It might be good to either raise an error or print a warning if the user passes in executor but the schemas aren't joinable/splittable. Right now it would just silently not actually used the initialized threadpool right?

Reply:
executor can be used for a general async run outside of dynamic batch